packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -178,8 +176,16 @@ contract MixinExchangeCore is
         // Validate the order
         require(order.makerAssetAmount > 0);
         require(order.takerAssetAmount > 0);
-        require(order.makerAddress == msg.sender);
 
+        // Validate sender
+        if (order.senderAddress != address(0)) {
+            require(order.senderAddress == msg.sender);
+        }
+        
+        // Validate transaction signed by maker
+        address makerAddress = getSignerAddress();
```

Comment:
Should we add something like this in `cancelOrdersUpTo`?

Reply:
`cancelOrdersUpTo` is tricky. Since it doesn't actually take an order, the sender is never validated. Basically, there would be no reason to ever call it via `executeTransaction` at the moment.

It is a bit weird that there is this constraint on `cancelOrder`, but that makers can still cancel via `cancelOrdersUpTo`.  I wonder of there's a way around this?